dash_agent/README.md

```diff
@@ -1,39 +1,174 @@
 
-TODO: List prerequisites and provide or point to information on how to
-start using the package.
+```shell
+dash-cli create {{agent}}
+```
 
 ## Usage
 
-TODO: Include short and useful examples for package users. Add longer examples
-to `/example` folder.
+This package contain the building blocks for creating dash agents:
+
+### AgentConfiguration
+
+The main part of the framework that glues together your agent configuration is `AgentConfiguration`. It is the base class for creating agent.
+
+Sample example of `AgentConfiguration` for demonstration purpose:
+
+```dart
+class MyAgent extends AgentConfiguration {
+  final docsSource = DocsDataSource();
+
+  @override
+  List<DataSource> get registeredDataSources => [docsSource];
+
+  @override
+  List<Command> get registerSupportedCommands =>
+      [AskCommand(docsSource: docsSource)];
+}
+```
+
+The above sample AgentConfiguration object taking data sources and supported commands. Both of them are explained below:
+
+- `DataSource`: Data sources will enable you to pass on any form of data that your agent might need to perform its intended tasks.
+- `Command`: Commands are the specialised tasks you want your agents to perform (like understanding the developer query, refactoring, code generation, etc).
+
+### Datasource
+
+As described above data sources lets you attach the data that your agent will need to perform its tasks. It can be anything from raw texts, json, files data, or even webpages.
+
+Sample example for DataSource for demostration purpose:
+
+```dart
+class DocsDataSource extends DataSource {
+
+ /// Enables you to pass data stored in files and directories in you local system.
+ @override
+ List<FileDataObject> get fileObjects => [
+       FileDataObject.fromFile(File(
+           'your_file_path')),
+        FileDataObject.fromDirectory(Directory(
+            'directory_path_to_data_source'))
+     ];
+ 
+
+ /// Enables you to pass in raw string and json data
+ @override
+ List<ProjectDataObject> get projectObjects =>
+     [ProjectDataObject.fromText('Data in form of raw text')];
+
+
+ /// Enables you to pass web data your agent by passing web pages urls or sitemaps in the object.
+  @override
+ List<WebDataObject> get webObjects =>
+     [WebDataObject.fromWebPage('https://sampleurl.com'), 
+     WebDataObject.fromSiteMap('https://sampleurl.com/sitemap.xml')];
+}
+```
+
+To create any data object, the easiest (and also the recommended way) is to call static functions of the above shared bases class of the object. For example if you want to store the json data. You can add the json data as shared below:
 
 ```dart
-const like = 'sample';
+final yourJson = {'key': 'data'};
+final jsonDataSource = ProjectDataSource.fromJson(yourJson);
 ```
 
+**Note**: At the moment, storing pdf files is not supported out of the box. However you can extract the relevant data from the pdf using the open source tools and pass the extracted data either via project data object or system data object.
```

Comment:
```suggestion
**Note**: At the moment, storing only text based files like code, markdown or raw text is only supported.
```

Reply:
@samyakkkk, I mentioned the note this way because if they come across a path where they don't know how to share the data. They can keep this reference in mind as a way to deal with them since PDF is one of the prominent ways of data archiving. I used it as an example. But open to your thoughts as well. 

Please commit this as you seem fit post reading it!

Reply:
@zexross I definitely think it's very thoughtful and should definitely be a part of the documentation. 

I just feel that at the readme level, it is not a must have piece of information.